meds/maker.py

```diff
@@ -1373,26 +1396,26 @@ def _load_config(self, config):
             self._joblib_threads = None
 
 
-def _psf_rec_func(output_path, psf_data, file_ids, rows, cols):
+def _psf_rec_func(output_path, psf_data, file_ids, rows, cols, colors):
     import joblib
 
-    joblib.dump(
-        [
-            psf_data[file_id].get_rec(row, col)
-            for file_id, row, col in zip(file_ids, rows, cols)
-        ],
-        output_path,
-    )
+    psfs = []
+    for file_id, row, col, color in zip(file_ids, rows, cols, colors):
+        if color is not None:
+            psf = psf_data[file_id].get_rec(row, col, color=color)
+        else:
+            psf = psf_data[file_id].get_rec(row, col)
+        psfs.append(psf)
+
+    joblib.dump(psfs, output_path)
+
     return output_path
 
 
 def _sky2image_func(wcs, ra, dec, color=None):
-
     if color is not None:
-        try:
-            res = wcs.sky2image(ra, dec, color=color)
-        except TypeError:
-            res = wcs.sky2image(ra, dec)
+        res = wcs.sky2image(ra, dec, color=color)
     else:
         res = wcs.sky2image(ra, dec)
+
```

Comment:
I changed the TypeError handling per your request. Note that I was following the pattern here in the WCS handling. I've changed that pattern here too.